packages/block-library/src/image/edit.js

```diff
@@ -475,10 +481,11 @@ class ImageEdit extends Component {
 								imageHeight,
 							} = sizes;
 
+							const defaultedAlt = alt ? alt : sprintf( __( 'This image has an empty alt attribute; its file name is %s' ), this.getFilename( url ) );
```

Comment:
The alternative if we don't trust the pattern match would be:

```suggestion
							const filename = this.getFilename( url );
							let defaultedAlt;
							if ( alt ) {
								defaultedAlt = alt;
							} else if ( filename ) {
								defaultedAlt = sprintf( __( 'This image has an empty alt attribute; its file name is %s' ), filename );
							} else {
								defaultedAlt = __( 'This image has an empty alt attribute' );
							}
```

Reply:
I feel a lot better about this suggested approach. At the very least we have a fallback in the edge case.